hollow/src/main/java/com/netflix/hollow/api/consumer/index/AbstractHollowHashIndex.java

```diff
@@ -94,16 +94,17 @@ public void detachFromDataRefresh() {
     }
 
     private class RefreshListener implements HollowConsumer.RefreshListener {
-        @Override public void deltaUpdateOccurred(HollowAPI api, HollowReadStateEngine stateEngine, long version) throws Exception {
-            reindex(stateEngine, api);
-        }
+        @Override
+        public void snapshotUpdateOccurred(HollowAPI refreshAPI, HollowReadStateEngine stateEngine, long version) {
+            idx.detachFromDeltaUpdates();
+            idx = new HollowHashIndex(stateEngine, queryType, selectFieldPath, matchFieldPaths);
```

Comment:
Yes, and so it will be for deltas too internally via the call to `HollowTypeStateListener.endUpdate` event.  For long delta chains the previous code would retain up a bunch of redundant HollowHashIndex instances all listening and doing the same work and possibly contributing to holding the lock longer than necessary.